### PR TITLE
[deepbook] Better errors, better borrow, more documentation, cleanups

### PR DESCRIPTION
**Pull request by damirka**
## Description 

Contains:

- cleanups
- better variable names
- errors cleanup and documentation
- better use of coin and balance APIs
- removes "returns" in many places

Deprecates (and removes all non-public members):

- clob (v1)
- custodial (v1)

Only micro-refactoring is adding a function: `all_orders_canceled_from_order` to reduce code duplication.

## Test Plan 

Current tests pass, no new features introduced.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move [hunk 1]
```diff
@@ -1183,14 +1256,16 @@ module deepbook::clob_v2 {
...
         if (restriction == IMMEDIATE_OR_CANCEL) {
             return (base_quantity_filled, quote_quantity_filled, false, 0)
         };
+
         if (restriction == FILL_OR_KILL) {
             assert!(base_quantity_filled == quantity, EOrderCannotBeFullyFilled);
             return (base_quantity_filled, quote_quantity_filled, false, 0)
         };
+
+        let order_id;
```
Comment (amnn) [inline crates/sui-framework/packages/deepbook/sources/clob_v2.move:1268]: Is there any point declaring this outside of the `if` -- it never escapes the respective branches.

File: crates/sui-framework/packages/sui-framework/tests/kiosk/kiosk_tests.move [hunk 2]
```diff
@@ -128,6 +128,8 @@ module sui::kiosk_tests {
         abort 1337
     }
 
+    #[test_only]
+    #[allow(unused_field)]
```
Comment (amnn) [inline crates/sui-framework/packages/sui-framework/tests/kiosk/kiosk_tests.move:132]: Unrelated?

**Review by amnn — APPROVED** [review]
Seems reasonable, barring nits!

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move [hunk 3]
```diff
@@ -348,21 +385,27 @@ module deepbook::clob_v2 {
...
         custodian::increase_user_available_balance(
             &mut pool.base_custodian,
             account_owner(account_cap),
             coin::into_balance(coin)
         );
+
         event::emit(DepositAsset<BaseAsset>{
             pool_id: *object::uid_as_inner(&pool.id),
-            quantity,
+            quantity: coin_amount,
```
Comment (sblackshear) [inline crates/sui-framework/packages/deepbook/sources/clob_v2.move:408]: Not sure if this change improves the code... field name punning makes the code shorter and easier to read

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move [hunk 4]
```diff
@@ -372,51 +415,65 @@ module deepbook::clob_v2 {
...
         custodian::increase_user_available_balance(
             &mut pool.quote_custodian,
             account_owner(account_cap),
             coin::into_balance(coin)
         );
+
         event::emit(DepositAsset<QuoteAsset>{
             pool_id: *object::uid_as_inner(&pool.id),
-            quantity,
+            quantity: coin_amount,
```
Comment (sblackshear) [inline crates/sui-framework/packages/deepbook/sources/clob_v2.move:429]: Same

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move [hunk 5]
```diff
@@ -986,9 +1057,9 @@ module deepbook::clob_v2 {
                 clock::timestamp_ms(clock),
                 coin::into_balance(quote_coin),
             );
-            join(
-                &mut base_coin,
-                coin::from_balance(base_balance_filled, ctx),
+            balance::join(
+                coin::balance_mut(&mut base_coin),
```
Comment (sblackshear) [inline crates/sui-framework/packages/deepbook/sources/clob_v2.move:1061]: Is there a `coin` API that lets you directly smash in a `balance` (and should we consider adding one if not)?

File: crates/sui-framework/packages/deepbook/sources/custodian.move [hunk 6]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Mysten Labs, Inc.
```
Comment (sblackshear) [inline crates/sui-framework/packages/deepbook/sources/custodian.move:1]: This code should no longer be used--we should delete everything that the upgrade policy will allow to avoid accidental future refactors/cleanups.

**Review by sblackshear — APPROVED** [review]

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move (hunk 1 above)
Comment (damirka) [reply crates/sui-framework/packages/deepbook/sources/clob_v2.move:1268]: It was this way, so I decided to not intervene much. Makes sense to patch!

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move (hunk 3 above)
Comment (damirka) [reply crates/sui-framework/packages/deepbook/sources/clob_v2.move:408]: But then it's easily confused with `quantity` that is used everywhere else.

File: crates/sui-framework/packages/deepbook/sources/clob_v2.move (hunk 5 above)
Comment (damirka) [reply crates/sui-framework/packages/deepbook/sources/clob_v2.move:1061]: We could consider it, yes.

File: crates/sui-framework/packages/deepbook/sources/custodian.move (hunk 6 above)
Comment (damirka) [reply crates/sui-framework/packages/deepbook/sources/custodian.move:1]: Should we abort everywhere with "not implemented"?

File: crates/sui-framework/packages/sui-framework/tests/kiosk/kiosk_tests.move (hunk 2 above)
Comment (damirka) [reply crates/sui-framework/packages/sui-framework/tests/kiosk/kiosk_tests.move:132]: Yup.

**Review by 666lcz — APPROVED** [review]
Thanks for the cleanup!

**Comment by damirka** [comment]
@healthydeve @666lcz closing this one.
